contrib/gendocs.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python3
+
+# This script generates a static documentation web site
+# by parsing the `**/*.asiidoc` files from the repository.
+#
+# Dependencies:
+# * Python 3
+# * `xdg-open` for opening the final result in a web browser.
+# * `antora` - a static documentation web site generator,
+#     https://docs.antora.org/antora/latest
+#
+# Usage:
+# ```console
+# $ ./contrib/gendocs.py
+# ```
+#
+# After running it should open the generated web site in a browser.
+#
+
+
+from dataclasses import dataclass
+from typing import Optional
+import atexit
+import glob
+import itertools
+import os
+import pathlib
+import re
+import shutil
+import subprocess
+
+# Get the script directory.
+script_dir = os.path.dirname(os.path.realpath(__file__))
+
+# Switch to the projects root dir.
+os.chdir(os.path.join(script_dir, ".."))
+
+
+# Revert the changes upon script exit.
+def cleanup():
+    try:
+        shutil.move("test/tests.asciidoc", "test/README.asciidoc")
+    except BaseException:
+        pass
+    try:
+        shutil.move("index.asciidoc", "README.asciidoc")
+    except BaseException:
+        pass
+    try:
+        shutil.move("VIMTOKAK.asciidoc", "VIMTOKAK")
+    except BaseException:
+        pass
+
+
+atexit.register(cleanup)
+
+
+# Antora fails if the repo contains broken symbolic links.
+shutil.rmtree("libexec", ignore_errors=True)
+
+# Canonical Antora paths.
+# See: https://docs.antora.org/antora/latest/standard-directories.
+#      https://docs.antora.org/antora/latest/root-module-directory.
+os.makedirs("modules/ROOT/images", exist_ok=True)
+os.makedirs("modules/ROOT/pages", exist_ok=True)
```

Comment:
FWIW instead of doing an in-tree build that requires renaming of some files,
I'd prefer to creating a separate directory (`html_doc`?),
copy all asciidoc files there and put all build artifacts there.
Not renaming files  avoids confusing editors etc.

Reply:
@krobelus Hey, long time. I've finally did that, now it keeps everything in the `doc_gen` dir without modifying the worktree.